src/content/plugins/normal-module-replacement-plugin.md

```diff
@@ -40,50 +41,78 @@ Conditional build depending on an [specified environment](/configuration/configu
 Say you want a configuration with specific values for different build targets.
 
 ``` javascript
-module.exports = function(env) {
-  var appTarget = env.APP_TARGET || 'VERSION_A';
+module.exports = function(env, args) {
   return {
     plugins: [
       new webpack.NormalModuleReplacementPlugin(/(.*)-APP_TARGET(\.*)/, function(resource) {
-        resource.request = resource.request.replace(/-APP_TARGET/, `-${appTarget}`);
+        var original = resource.request;
+        var replace = resource.request.replace(/-APP_TARGET/, `-${env}`);
+        resource.request = replace;
+        // This will help you debug the replacement during the build process.
+        console.log(`Original Config: '${original}', Replacement: '${replace}'`);
       })
     ]
   };
 
 };
 ```
 
-Create the two config files:
+Create the two config files (the conventions are the same for JavaScript and TypeScript):
 
-__app/config-VERSION_A.js__
+__config/config-BETA.js (OR .ts)__ 
 
 ``` javascript
 export default {
-  title : 'I am version A'
+  title : 'I am the Beta configuration.'
 };
 ```
 
-__app/config-VERSION_B.js__
+__config/config-PROD.js (OR .ts)__
 
 ``` javascript
 export default {
-  title : 'I am version B'
+  title : 'I am the Production configuration.'
 };
 ```
 
 Then import that config using the keyword you're looking for in the regexp:
 
 ``` javascript
-import config from 'app/config-APP_TARGET';
+import config from './config/config-APP_TARGET';
 console.log(config.title);
 ```
 
-And now you just get the right config imported depending on which target you're building for:
+And now the right config will be imported depending on which target you're building for:
 
 ``` shell
-webpack --env.APP_TARGET VERSION_A
-=> 'I am version A'
+webpack --mode production --env BETA
+=> 'I am the Beta configuration.'
 
-webpack --env.APP_TARGET VERSION_B
-=> 'I am version B'
+webpack --mode production --env PROD
+=> 'I am the Production configuration.'
 ```
+
+
+## TypeScript Considerations
+
+The TypeScript compiler requires that all files be physically-present in order to generate an output. If you attempt to
+complete the above tasks, you will likely receive the following error:
+
+``` shell
+TS2307: Cannot find module './config/config-APP_TARGET'.
+```
+
+To correct this issue, you'll need to create an additional file to satisfy the compiler without breaking the
```

Comment:
Doesnt sound right to me, suggest to change `To correct this issue` => `In order to make it work, ...`

Reply:
I called it an "issue" because the pattern works in JavaScript and not in the TypeScript compiler, and the compiler throws an error. I'm opening an issue in the TypeScript project to see if there is a way to resolve it without my hack... at which point we could take out this whole section.

Happy to make the change if you still feel it's necessary.

Reply:
When i am reading "to correct an issue" i get confused, i had no issues and was following a guide, maybe we need more eyes on this cc @montogeek

Reply:
I've made changes to this as requested.